packages/apollo-cli/src/commands/codegen/generate.ts

```diff
@@ -0,0 +1,209 @@
+import 'apollo-codegen-core/lib/polyfills';
+import { Command, flags } from "@oclif/command";
+import * as Listr from "listr";
+import * as path from "path";
+
+import { TargetType, default as generate } from '../../generate';
+
+import { buildClientSchema } from "graphql";
+
+import * as globby from "globby";
+import * as fs from 'fs';
+import { promisify } from 'util';
+
+import { toPromise, execute } from "apollo-link";
+
+import { engineLink, getIdFromKey } from "../../engine";
+import { SCHEMA_QUERY } from "../../operations/schema";
+
+export default class Generate extends Command {
+  static description = "Generate static types for GraphQL queries.";
+
+  static flags = {
+    help: flags.help({
+      char: "h",
+      description: "Show command help",
+    }),
+    queries: flags.string({
+      description: "Path to your GraphQL queries, can include search tokens like **",
+      default: "**/*.graphql",
+    }),
+    schema: flags.string({
+      description: "Path to your GraphQL schema introspection result",
+      default: "schema.json",
+    }),
+
+    key: flags.string({
+      description: "The API key for the Apollo Engine service",
+    }),
+    tag: flags.string({
```

Comment:
lets pull back tag for right now since we don't have that concept in Engine quite yet